### PR TITLE
[ITS] Service category breadcrumbs

### LINKED ISSUE #6872: Show category in breadcrumb on services
# Story

As an ITS website owner, I would like to output the category associated with a service in the breadcrumb on the service page so that users can click the link to view all services that are associated with that particular category.

# Additional Info

The category field exists on the service content type but is not currently being output anywhere

We should output the category in the breadcrumb as a link to that term page.

At this point we don't need to output the category anywhere else on the service.

# Questions

Do we need to output the category in the breadcrumb? - Yes
Do we need the category to be reflected in the url alias? - No

There is some internal conversation around these questions https://github.com/uiowa/uiowa/pull/6821#issuecomment-1703245392

# Proposed solution

- Output breadcrumbs on services - Home - Services - Service Category - Title of service

### PR DESCRIPTION
**Pull request by cory-skeers**
Resolves #6872 

# How to test

`ddev blt frontend`

```
ddev blt ds --site=its.uiowa.edu && ddev drush @its.local config:import --source ../config/sites/its.uiowa.edu --partial -y && ddev drush @its.local cr && ddev drush @its.local uli
```

Add a Page w/title "Services" (specifically, a URL of `/services`). This'll be the overall landing page for Services.

Add the service with no Service Category, check out the breadcrumbs. It should be `Home / Services / <title>`
Add a Service Category taxonomy term, update a Service with that Service Category, and check out the breadcrumbs. It should be ``Home / Services / <service category> / <title>`, with appropriate links.
Check that the Service Category term page URL is of the form `services/<term-name>`

Check other node types and pages (taxonomy terms, views, etc) and that their breadcrumbs are unaffected

### CODE REVIEW

**Comment by joewhitsitt** [comment]
In the moment I think I said it was our pattern to include the current page title as part of the breadcrumb. That is true our menu based breadcrumbs, but isn't true for anything else.

https://classrooms.uiowa.edu/abw-110
https://housing.uiowa.edu/residence-halls/catlett
https://admissions.uiowa.edu/academics/accounting
https://grad.uiowa.edu/academics/thesis-defenses/2023-10-02/stephanie-peterson
https://sandbox.prod.drupal.uiowa.edu/leadership-team/sean-adams-hiett

At this point, I'd be in-favor of removing it.

https://github.com/uiowa/uiowa/pull/6984/files#diff-2f46a41d957b6d80db1efdb16c51c9d5b134ef38942e954693c24d9be8f0a83cR233-R237

**Comment by joewhitsitt** [comment]
pulling to test

File: docroot/sites/its.uiowa.edu/modules/its_core/its_core.module [hunk 1]
```diff
@@ -210,3 +207,32 @@ function its_core_preprocess_views_view_unformatted(&$variables) {
...
+        'text' => $category_name,
+        'url' => $category_path,
+      ];
+    }
+  }
+  // Also include an unlinked crumb to the current page
+  // matching our breadcrumb pattern.
+  $variables['breadcrumb'][] = [
+    'text' => $node->getTitle(),
+  ];
```
Comment (cory-skeers) [inline docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: I'm fine with removing this, matching non-menu-based breadcrumbs.

```suggestion
```

Comment (joewhitsitt) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: @briand44 are you okay with this based on https://github.com/uiowa/uiowa/pull/6984#issuecomment-1756150733 ?

Comment (joewhitsitt) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: oops. github notification delay. Cory was too fast!

**Comment by joewhitsitt** [comment]
Missing `ddev drush @its.local en its_core` from the test instructions. After a head scratching moment... Onward!

**Review by joewhitsitt — APPROVED** [review]
This works as expected. Not our first time using preprocess_breadcrumb. Even handles funky characters.

Screenshot 2023-10-10 at 3 35 56 PM

File: docroot/sites/its.uiowa.edu/modules/its_core/its_core.module (hunk 1 above)
Comment (briand44) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: umm... I would like to understand a little more. Will there be other examples of breadcrumbs on ITS without the node title?

Comment (joewhitsitt) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: articles for one: https://sandbox.prod.drupal.uiowa.edu/news/2023/02/joes-new-article-test

unless it lives in the menu system, it doesn't get the content title appended. i imagine support articles, software be the same unless custom breadcrumb work will be done there too.

Comment (cory-skeers) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: Articles: 
image

And and pages that aren't placed directly in the menu:
image

Pages that are placed in the menu get the page title added to the breadcrumb as part of that
image

Comment (briand44) [reply docroot/sites/its.uiowa.edu/modules/its_core/its_core.module]: Thanks, yeah I think that makes sense then.

**Pull request merged**